### PR TITLE
add sidebar display to settings page

### PR DESCRIPTION
**Pull request by SDavisMedia**
Resolves: #25

What I've done here is...

1. Make slight tweaks to the Settings page HTML in order to handle content/sidebar as well as full content.

2. Added the ability to display a sidebar on the Settings page, based on a time range down to the second. To see this in testing, adjust _line 511_ in _sugar-event-calendar/includes/admin/settings.php_...

from `$start = strtotime( '2019-11-29 06:00:00' )`
to `$start = strtotime( '2019-11-04 06:00:00' )`

**Notes:**

Markup additions to the Settings page are RTL-ready. Text is translation-ready, though it probably won't ever be translated. Overall, this kind of feels like a temporary system since the content is obviously outdated in a month or so. But I believe the _correct_ approach involves a call home. And of course, we can't manage that right now.

### CODE REVIEW

File: sugar-event-calendar/includes/admin/settings.php [hunk 1]
```diff
@@ -482,3 +499,67 @@ function display_subsection() {
...
+ */
+function maybe_display_sidebar() {
+
+	// Set the date/time range based on UTC
+	$start = strtotime( '2019-11-29 06:00:00' );
+	$end   = strtotime( '2019-12-07 05:59:59' );
+	$now   = time();
+
+	// Only display sidebar if the page is loaded within the date range
+	if ( $now > $start && $now < $end ) {
```
Comment (JJJ) [inline sugar-event-calendar/includes/admin/settings.php]: Recommend to wrap comparisons in their own parenthesis:

`( ( $now > $start ) && ( $now < $end ) )`

File: sugar-event-calendar/includes/admin/settings.php [hunk 2]
```diff
@@ -482,3 +499,67 @@ function display_subsection() {
...
+function display_sidebar() {
+	$coupon_code = 'BFCM2019';
+	?>
+
+	<div class="sc-settings-sidebar">
+
+		<div class="sc-settings-sidebar-content">
+
+			<div class="sc-sidebar-header-section">
+				<img class="bcfm-header" src="<?php echo SC_PLUGIN_URL . 'includes/admin/assets/images/bfcm-header.svg'; ?>">
```
Comment (JJJ) [inline sugar-event-calendar/includes/admin/settings.php]: Recommend to wrap the `src` in `esc_url()`:

`echo esc_url( SC_PLUGIN_URL . 'includes/admin/assets/images/bfcm-header.svg' );`

File: sugar-event-calendar/includes/admin/settings.php [hunk 3]
```diff
@@ -482,3 +499,67 @@ function display_subsection() {
...
+
+			<div class="sc-sidebar-footer-section">
+				<a class="cta-button" href="https://sugarcalendar.com/pricing/"><?php _e( 'Upgrade Now!', 'sugar-calendar' ); ?></a>
+			</div>
+
+		</div>
+
+		<div class="sc-sidebar-logo-section">
+			<div class="sc-logo-wrap">
+				<img class="sc-logo" src="<?php echo SC_PLUGIN_URL . 'includes/admin/assets/images/sugar-calendar-logo-light.svg'; ?>">
```
Comment (JJJ) [inline sugar-event-calendar/includes/admin/settings.php]: See above about using `esc_url()`.

File: sugar-event-calendar/includes/admin/settings.php [hunk 4]
```diff
@@ -378,15 +381,29 @@ function page() {
...
 
-			<?php submit_button(); ?>
+				</form>
 
-			<?php settings_fields( "sc_{$section}_{$subsection}" ); ?>
+			</div>
 
-		</form>
+			<?php
+			if ( $maybe_display_sidebar ) {
```
Comment (JJJ) [inline sugar-event-calendar/includes/admin/settings.php]: Recommend to be explicit:

`if ( true === $maybe_display_sidebar ) {`

File: sugar-event-calendar/includes/admin/settings.php [hunk 5]
```diff
@@ -378,15 +381,29 @@ function page() {
 
 		<hr class="wp-header-end">
 
-		<form method="post" action="options.php">
+		<div class="sc-settings-main<?php echo $maybe_display_sidebar ? ' has-sidebar' : ''; ?> wp-clearfix">
```
Comment (JJJ) [inline sugar-event-calendar/includes/admin/settings.php]: Recommend to make this a variable underneath line 365:

```
$wrapper_class = ( true === $maybe_display_sidebar )
	? ' has-sidebar'
	: '';
```

Then:
```
<div class="sc-settings-main<?php echo esc_attr( $wrapper_class ); ?> wp-clearfix">
```

File: sugar-event-calendar/includes/admin/assets/images/sugar-calendar-logo-light.svg [hunk 6]
```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
```
Comment (JJJ) [inline sugar-event-calendar/includes/admin/assets/images/sugar-calendar-logo-light.svg]: cc @amdrew for consult: any idea if this SVG needs to be compressed?

I have a fuzzy memory about omitting doctypes and rounding values to reduce file size, but I could be mistaken.

File: sugar-event-calendar/includes/admin/assets/images/bfcm-header.svg [hunk 7]
```diff
@@ -0,0 +1,115 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd">
```
Comment (JJJ) [inline sugar-event-calendar/includes/admin/assets/images/bfcm-header.svg]: cc @amdrew : see comment below about SVG compression

File: sugar-event-calendar/includes/admin/assets/images/sugar-calendar-logo-light.svg (hunk 6 above)
Comment (SDavisMedia) [reply sugar-event-calendar/includes/admin/assets/images/sugar-calendar-logo-light.svg]: @JJJ I can definitely optimize the SVG. While working on this, I exported the SVG in several different ways, and also embedded it in several different ways. I couldn't decide how much CSS control I wanted, if I wanted to go inline or `img`, etc. Eventually I just stopped obsessing and moved on with the project.

Now's a good time to go back and optimize. So thanks for the reminder! All of the other suggestions are perfect. Gonna hit everything.

Thank you sir!

**Comment by SDavisMedia** [comment]
@JJJ That should take care of everything. Notice I changed a bit of the design. Part of that effort was to make it shorter, which I think should be in check now. Another part of it, though, was that I received some intended copy from the marketing folks. That copy was shorter. So that was good timing. Because of all that, I ended up centering all text (normally not my cup of tea). The result _should be_ solid on all screens.

Image 2019-11-05 at 9 01 20 AM

**Comment by SDavisMedia** [comment]
Another quick note, please don't merge this. There will be tweaks coming to things like class names, copy, etc. Next time I'll wait to request review. I got some critiques a little late in the game.

**Comment by SDavisMedia** [comment]
Okay, @JJJ! Calling this "done" in its current state. I didn't add any other promos. I looked into it for a few but realized I'd have to learn a bit about how the code is written, and I don't have time at the moment. I was intrigued, but I have to keep pace.

If there's anything you want to add before going live, that would be cool. I also think moving forward as-is is cool. So your call there!

This is ready for a final review. If everything meets the standard, you can take over from here and release how you see fit!

Thanks for all the help. 🙌

**Pull request merged**